lib/backend/k8s/resources/node_conversion.go

```diff
@@ -49,9 +55,31 @@ func K8sNodeToCalico(node *kapiv1.Node) (*model.KVPair, error) {
 			FelixIPv4:   ip,
 			Labels:      node.Labels,
 			BGPIPv4Addr: ip,
-			BGPIPv6Addr: &net.IP{},
-			BGPASNumber: &asn,
+			BGPASNumber: &bgpAsn,
+			BGPIPv4Net:  bgpV4Cidr,
 		},
 		Revision: node.ObjectMeta.ResourceVersion,
 	}, nil
 }
+
+// Convert a Calico Node to Kubernetes, place BGP configuration info in annotations
+func CalicoToK8sNode(kvp *model.KVPair) (*kapiv1.Node, error) {
+	vals := kvp.Value.(model.Node)
+	annotations := map[string]string {
+		"projectcalico.org/BGPIPv4Addr": vals.BGPIPv4Addr.String(),
+		"projectcalico.org/BGPASNumber": vals.BGPASNumber.String(),
+		"projectcalico.org/BGPIPv4Net":  vals.BGPIPv4Net.String(),
+	}
+
+	nodeMeta := kapiv1.ObjectMeta{
+		Name:        kvp.Key.(model.NodeKey).Hostname,
+		Annotations: annotations,
+		Labels:      vals.Labels,
+	}
+
+	node := &kapiv1.Node{
```

Comment:
Hmm, I don't think this is going to work.

We're returning a pretty much empty Node object.  When we write this to the API, it's probably going to delete a whole bunch of information on the Node.

For example, there's no `Spec` here.  Also, what happens if there are other annotations on the Node?

I think, unfortunately, we need to make the Update() method do a Get() from the k8s API for this node, then populate the fields we care about and write it back to the API. There's not enough information in a Calico "Node" resource to regenerate the entire Kubernetes "Node" object.